src/libmongoc/CMakeLists.txt

```diff
@@ -1345,6 +1345,11 @@ endif ()
 
 include (CMakePackageConfigHelpers)
 
+# Check if BCryptDeriveKeyPBKDF2 is defined in bcrypt.h
+if (WIN32 AND MONGOC_ENABLE_CRYPTO_CNG)
+   check_symbol_exists (BCryptDeriveKeyPBKDF2 bcrypt.h MONGOC_HAVE_BCRYPT_PBKDF2)
```

Comment:
`MONGOC_HAVE_BCRYPT_PBKDF2` is defined as a CMake variable, but is not-yet applied to the C code.

Update `mongoc-config.h.in` with [these instructions](https://github.com/mongodb/mongo-c-driver/blob/46ceebacf6b36683aa20f2ba90db1ebd2ac55b8f/src/libmongoc/src/mongoc/mongoc-config.h.in#L27-L35). `mongoc-config.h.in` is configured by CMake and to generate `mongoc-config.h`. CMake variables (between `@`) are substituted.

Reply:
I pushed new updates for this but I think I might have missed something. In line 73 of `mongoc-config.h` (`#if MONGOC_HAVE_BCRYPT_PBKDF2 != 1`) I see the error "operator has no left operand" which I believe means the macro is not defined?

Reply:
Try moving this block above `configure_file` on line 725, which generates the `mongoc-config.h` file:

```c++
configure_file (
   "${PROJECT_SOURCE_DIR}/src/mongoc/mongoc-config.h.in"
   "${PROJECT_BINARY_DIR}/src/mongoc/mongoc-config.h"
)
```